cpp/src/randomforest/randomforest_impl.cuh

```diff
@@ -425,18 +536,31 @@ void rfRegressor<T>::fit(const cumlHandle& user_handle, T* input, int n_rows,
          used to build the bootstrapped sample. Expectation: Each tree node will contain
          (a) # n_sampled_rows and (b) a pointer to a list of row numbers w.r.t original data.
     */
-
     DecisionTree::TreeMetaDataNode<T, T>* tree_ptr = &(forest->trees[i]);
-    trees[i].fit(user_handle, input, n_cols, n_rows, labels,
-                 sorted_selected_rows.data(), n_sampled_rows, tree_ptr,
-                 this->rf_params.tree_params, tempmem);
+    trees[i].fit(local_handle[stream_id], input, n_cols, n_rows, labels, rowids,
```

Comment:
Same thread safety concern as above.

Reply:
Decision Tree needs an handle because its its own separate cuML algo.
Two options
1. Remove cuml handle. But we wont be able to use DT on its own.
2. Over ride the .fit() to accept both cases. as such.

Reply:
I think option #2 sounds like a good solution. Maybe even have the decision tree `fit(cumlHandle)` call the fit(deviceAllocator, cudaStream_t)` version just to minimize duplication.

Reply:
Another option would be to make the whole of internals of DT, an ml-prim on its own.

Then, we can use this prim from the DT algo in cuML itself, as well as call the relevant ones from RF. Again, I'd say that this is better handled in a separate PR.

Reply:
@cjnolet The handle is passed thru the entire code chain. The change might not be straight forward. I can do it in a new PR once this is merged ? If thats okay.

Otherwise. I will do it in this one.

Reply:
I just checked the new algorithm here does not need handle. So in future PR when we remove the old algorithm, this change can be done

Reply:
> Another option would be to make the whole of internals of DT, an ml-prim on its own.

I like this option very much. 

@vishalmehta1991, I'm okay doing this in a follow-on PR.